### DISCUSSION TITLE
Password lost

### DISCUSSION THREAD

**Discussion by Kubis10** [xEdziu/KeepMyPassword-Desktop#28] [Q&A] [answered]
## What happens if you forget your app password?
This should be secured because people often forget their passwords

**Comment by xEdziu** [comment, marked as answer]
## You're doomed.

There is no way to recover your password.

## Why I will not implement this?

In terms of how registrating works, there is no real way to recover your password, once you create config file.

Important thing is that app **does not store your password**. Algorythm uses your password as a key to encrypt and than hash some initial word. Then this hash is compared to hash that was created while registrating. If those two are equal, you are logged in.

So in long term - there is no option to restore your password, after you lost it, because I (neither the app) don't know it - and this is intended behaviour.

## So what else can I do?
I am really sorry to say that, but only option to log into your app is to remember your password. Otherwise, all your data will stay encrypted and there is no way to recover it.

**Reply by xEdziu** [reply]
This is the same reason, why you won't be able to change your password. Changing your password to app in general will be the same thing as creating new account. You still will not have access to your already encrypted data and your new password will not match the key, that was generated from your previous password that you've been using.

**Reply by xEdziu** [reply]
I am aware that this policy is very strict, but this is the best way to keep data REALLY secure.
